lib/CamDiag.py

```diff
@@ -366,18 +394,30 @@ def create_climo(self, baseline=False):
         if cam_climo_dict['calc_cam_climo']:
 
             #If so, then extract names of time-averaging scripts:
-            avg_func_names = self.__time_averaging_scripts
-
+            avg_func_names = self.__time_averaging_scripts  # this is a list of script names 
+                                                            # _OR_ 
+                                                            # a **list** of dictionaries with script names as keys that hold args(list), kwargs(dict), and module(str)
+            
             #Extract necessary variables from CAM configure dictionary:
             input_ts_loc    = cam_climo_dict['cam_ts_loc']
             overwrite_climo = cam_climo_dict['cam_overwrite_climo']
             var_list        = self.__diag_var_list
 
             #Loop over all averaging script names:
             for avg_func_name in avg_func_names:
-
-                #Add file suffix to script name (to help with the file search):
-                avg_script = avg_func_name+'.py'
+                # option to specify the module's name:
+                if isinstance(avg_func_name, dict):
+                    assert len(avg_func_name) == 1  # must just be {function_name : {args:[...], <kwargs:{...}>, <module:'xxxx'>}}
```

Comment:
Can you add an error message like so:

`assert len(avg_func_name) == 1, "YAML function line must just be {function_name : {args:[...], <kwargs:{...}>, <module:'xxxx'>}}"`